pkg/prometheus/statefulset.go

```diff
@@ -86,6 +86,12 @@ func makeStatefulSet(p v1alpha1.Prometheus, old *v1beta1.StatefulSet, config *Co
 		},
 		Spec: *spec,
 	}
+
+	if p.Spec.ImagePullSecrets != nil && len(p.Spec.ImagePullSecrets) > 0 {
+		statefulset.Spec.Template.Spec.ImagePullSecrets = make([]v1.LocalObjectReference, len(p.Spec.ImagePullSecrets))
```

Comment:
Is there a specific reason for the copying here that I'm missing? If not just do 

```
statefulset.Spec.Template.Spec.ImagePullSecrets = p.Spec.ImagePullSecrets
```